src/eth_provider/database/ethereum.rs

```diff
@@ -356,4 +356,164 @@ mod tests {
             Some(mock_pending_transaction.tx)
         );
     }
+
+    fn u64_to_b256(value: u64) -> B256 {
+        let mut bytes = [0u8; 32];
+        bytes[24..32].copy_from_slice(&value.to_be_bytes());
+        B256::from(bytes)
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_ethereum_block_store() {
+        // Initialize MongoDB fuzzer
+        let mut mongo_fuzzer = MongoFuzzer::new(RANDOM_BYTES_SIZE).await;
+
+        // Start MongoDB Docker container
+        let _c = DOCKER_CLI.run(mongo_fuzzer.mongo_image());
+
+        // Mock a database with 100 transactions, receipts, and headers
+        let database = mongo_fuzzer.mock_database(100).await;
+
+        // Test fetching existing and none existing header via blockhash and blocknumber from database
+        test_get_header(&database, &mongo_fuzzer).await;
+
+        // Test fetching existing and none existing block via blockhash and blocknumber from database
+        test_get_blocks(&database, &mongo_fuzzer).await;
+
+        // Test fetching existing and none existing transaction counts via blockhash and blocknumber from database
+        test_get_transaction_count(&database, &mongo_fuzzer).await;
+    }
+
+    async fn test_get_header(database: &Database, mongo_fuzzer: &MongoFuzzer) {
+        let header_block_hash = mongo_fuzzer
+            .documents()
+            .get(&CollectionDB::Headers)
+            .unwrap()
+            .first()
+            .unwrap()
+            .extract_stored_header()
+            .unwrap()
+            .header
+            .clone();
+
+        let first_block_hash = header_block_hash.hash.unwrap();
+
+        // Test retrieving header by block hash
+        assert_eq!(database.header(first_block_hash.into()).await.unwrap().unwrap(), header_block_hash);
+
+        let first_block_number = header_block_hash.number.unwrap();
+
+        // Test retrieving header by block number
+        assert_eq!(database.header(first_block_hash.into()).await.unwrap().unwrap(), header_block_hash);
+
+        let false_block_hash = BlockHashOrNumber::from(u64_to_b256(425));
+
+        // Test retrieving non-existing header by block hash
+        assert_eq!(database.header(false_block_hash).await.unwrap(), None);
+
+        let false_block_number = BlockHashOrNumber::from(425);
+
+        // Test retrieving non-existing header by block number
+        assert_eq!(database.header(false_block_number).await.unwrap(), None);
+    }
+
+    async fn test_get_blocks(database: &Database, mongo_fuzzer: &MongoFuzzer) {
```

Comment:
I'm not a fan of this test, as it doesn't really test anything, it's just a copy paste of the code we already have in the `block` implementation of the `EthereumBlockStore`. Let's just keep the parts where you check that the returned value is `None`, while for the other values we can check for example that the returned value `is_some` and that the body field contains transactions.

Reply:
Agreed, the goal is to test all the combinations here:
- full = false / unknown block hash
- full = false / unknown block number
- full = true / unknown block hash
- full = true / unknown block number
- What if the `withdrawals_root` is not `EMPTY_ROOT_HASH`, we should raise an error
- What if no transactions
- ....

Reply:
> Agreed, the goal is to test all the combinations here:
> 
>     * full = false / unknown block hash
> 
>     * full = false / unknown block number
> 
>     * full = true / unknown block hash
> 
>     * full = true / unknown block number
> 
>     * What if the `withdrawals_root` is not `EMPTY_ROOT_HASH`, we should raise an error
> 
>     * What if no transactions
> 
>     * ....

I'm not sure what you mean by test if there are no transactions as I can't see immediately where an error would pop up if a block is published without any transctions, (both eth and starknet blocks can be published without transactions right?)

Reply:
You could check that the transactions in the block are empty in this case to ensure that the function handles that properly